app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class SessionsController < ApplicationController
+  layout 'auth', only: %i[new create]
+
+  before_action :redirect_to_home, only: %i[new create]
+
+  def new; end
+
+  def create
+    user = User.find_by(username: params[:username])
+
+    if user&.authenticate(params[:password])
+      session[:user_id] = user[:id]
+
+      redirect_to root_path
```

Comment:
and if we agree with the above comment, this line can just be `redirect_to_home`

Reply:
Fixed in 9b7412ae9dadd49b297cc23af7dd8494a98a9731